gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/task/HiveConverterUtils.java

```diff
@@ -0,0 +1,437 @@
+      dmlQuery.append(String.format("INSERT OVERWRITE TABLE `%s`.`%s` %n", outputDbName, outputTblName));
+    } else {
+      dmlQuery.append(String.format("INSERT INTO TABLE `%s`.`%s` %n", outputDbName, outputTblName));
+    }
+
+    // Partition details
+    dmlQuery.append(partitionKeyValues(optionalPartitionDMLInfo));
+
+    // If not exists
+    if (shouldCreateIfNotExists) {
+      dmlQuery.append(" IF NOT EXISTS \n");
+    }
+    dmlQuery.append(String.format("SELECT * FROM `%s`.`%s`", inputDbName, inputTblName));
+    if (optionalPartitionDMLInfo.isPresent()) {
+      if (optionalPartitionDMLInfo.get().size() > 0) {
+        dmlQuery.append(" WHERE ");
+        boolean isFirstPartitionSpec = true;
+        for (Map.Entry<String, String> partition : optionalPartitionDMLInfo.get().entrySet()) {
+          if (isFirstPartitionSpec) {
+            isFirstPartitionSpec = false;
+          } else {
+            dmlQuery.append(" AND ");
+          }
+          dmlQuery.append(String.format("`%s`='%s'",
+              partition.getKey(), partition.getValue()));
+        }
+        dmlQuery.append(" \n");
+      }
+    }
+    return dmlQuery.toString();
+    //dmlQuery.append(String.format(" %n FROM `%s`.`%s` ", inputDbName, inputTblName));
+  }
+
+  protected static StringBuilder partitionKeyValues(Optional<Map<String, String>> optionalPartitionDMLInfo) {
+    StringBuilder partitionKV = new StringBuilder();
+    if (optionalPartitionDMLInfo.isPresent()) {
+      if (optionalPartitionDMLInfo.get().size()  > 0) {
+        partitionKV.append("PARTITION (");
+        boolean isFirstPartitionSpec = true;
+        for (Map.Entry<String, String> partition : optionalPartitionDMLInfo.get().entrySet()) {
+          if (isFirstPartitionSpec) {
+            isFirstPartitionSpec = false;
+          } else {
+            partitionKV.append(", ");
+          }
+          partitionKV.append(String.format("`%s`", partition.getKey()));
+        }
+        partitionKV.append(") \n");
+      }
+    }
+    return partitionKV;
+  }
+
+  public static void cleanUpNonPartitionedTable(Map<String, String> publishDirectories, List<String> cleanupQueries,
+      String stagingDataLocation, List<String> cleanupDirectories, String outputDataLocation,
+      String outputTableDatabase, String stagingTableName) {
+    log.debug("Snapshot directory to move: " + stagingDataLocation + " to: " + outputDataLocation);
+    publishDirectories.put(stagingDataLocation, outputDataLocation);
+
+    String dropStagingTableDDL = HiveAvroORCQueryGenerator.generateDropTableDDL(outputTableDatabase, stagingTableName);
+
+    log.debug("Drop staging table DDL: " + dropStagingTableDDL);
+    cleanupQueries.add(dropStagingTableDDL);
+
+    log.debug("Staging table directory to delete: " + stagingDataLocation);
+    cleanupDirectories.add(stagingDataLocation);
+  }
+
+  public static void moveDirectory(FileSystem fs, String sourceDir, String targetDir) throws IOException {
```

Comment:
Why do we need a method for this? If needed, it should be in `HadoopUtils` or `FileSystemUtils`. It has nothing to do with Hive.